src/modm/driver/display/ssd1306_impl.hpp

```diff
@@ -10,64 +10,103 @@
 // ----------------------------------------------------------------------------
 
 #ifndef MODM_SSD1306_HPP
-#	error	"Don't include this file directly, use 'ssd1306.hpp' instead!"
+#error "Don't include this file directly, use 'ssd1306.hpp' instead!"
 #endif
 
-template < class I2cMaster, uint8_t Height >
-modm::Ssd1306<I2cMaster, Height>::Ssd1306(uint8_t address)
-:	I2cDevice<I2cMaster, 2, ssd1306::DataTransmissionAdapter<Height>>(address),
-	commandBuffer{0x80, 0, 0x80, 0, 0x80, 0, 0x80, 0, 0x80, 0, 0x80, 0, 0x80, 0}
-{
-	this->transaction.setCommandBuffer(commandBuffer);
-}
+template<class I2cMaster, uint8_t Height>
+modm::Ssd1306<I2cMaster, Height>::Ssd1306(uint8_t address) : I2cDevice<I2cMaster, 2, ssd1306::Ssd1306_I2cWriteTransaction>(address)
+{}
 
 // ----------------------------------------------------------------------------
 // MARK: - Tasks
-template < class I2cMaster, uint8_t Height >
+template<class I2cMaster, uint8_t Height>
 modm::ResumableResult<bool>
 modm::Ssd1306<I2cMaster, Height>::initialize()
 {
 	RF_BEGIN();
+	TransactionSuccess = true;
+
+	commandBuffer[0] = FundamentalCommands::DisplayOff;
+	commandBuffer[1] = TimingAndDrivingCommands::DisplayClockDivideRatio;
+	commandBuffer[2] = 8 << 4;  // Frequency (influences scrolling speed too)
+	commandBuffer[2] |= 0;      // Prescaler
+	commandBuffer[3] = HardwareConfigCommands::MultiplexRatio;
+	commandBuffer[4] = 63;
+	commandBuffer[5] = HardwareConfigCommands::DisplayOffset;
+	commandBuffer[6] = 0;
+	TransactionSuccess &= RF_CALL(writeCommands(7));
+
+	RF_CALL(initializeMemoryMode());
+
+	commandBuffer[0] = TimingAndDrivingCommands::ChargePump;
+	commandBuffer[1] = ChargePump::V7_5;
+	commandBuffer[2] = HardwareConfigCommands::SegmentRemap127;
+	commandBuffer[3] = HardwareConfigCommands::ComOutputScanDirectionDecrement;
+	commandBuffer[4] = HardwareConfigCommands::DisplayStartLine;
+	commandBuffer[4] |= 0;
```

Comment:
?

Reply:
It's to say hopefully self-explaining, an integer-nibble could be set here. In this case is 0.